pkg/alertmanager/multitenant_test.go

```diff
@@ -1161,10 +1159,22 @@ func TestAlertmanager_StateReplicationWithSharding(t *testing.T) {
 
 			// With sharding enabled, we propagate messages over gRPC instead of using a gossip over TCP.
 			// 1. First, get a random multitenant instance
-			multitenantAM := instances[rand.Intn(len(instances))]
+			//    We must pick an instance which actually has a user configured.
+			var multitenantAM *MultitenantAlertmanager
+			for {
+				multitenantAM = instances[rand.Intn(len(instances))]
+
+				multitenantAM.alertmanagersMtx.Lock()
+				amount := len(multitenantAM.alertmanagers)
+				multitenantAM.alertmanagersMtx.Unlock()
+				if amount > 0 {
+					break
+				}
+			}
 
 			// 2. Then, get a random user that exists in that particular alertmanager instance.
 			multitenantAM.alertmanagersMtx.Lock()
+			require.Greater(t, len(multitenantAM.alertmanagers), 0)
```

Comment:
If there's a case where my above comment holds true, then I think all we need to do is wait to use `Eventually` here to wait until this alertmanager has the number of replicas we expect. Depending on the test case it should be deterministic.